### PR TITLE
feature: Add screenshots to photos crud scenario

### PR DESCRIPTION
**Pull request by CozyKourai**
Photos crud scenario was already split into 3 scenario : Upload photos, photos crud, and Delete Photos.
For each one:
- Add const for test names and logs
- change `initVR` to `ctx.vr.start()`
- change name for existing screenshots
- add new screenshots
- move `checkAllImagesExists` to avoid error when reloading page
- change test `open random photos, not 1st nor last` to `open 2nd photo` , beaucse we need to open the same photos each time when taking screenshots

### CODE REVIEW

**Comment by CozyKourai** [comment]
[(𝐗𝐒) SCN - Photos CRUD - ajout screenshots](https://trello.com/c/SVzUdtob/1853-%F0%9D%90%97%F0%9D%90%92-scn-photos-crud-ajout-screenshots)

**Comment by Crash--** [comment]
@CozyKourai conflit à corriger

File: testcafe/tests/photos/photos_crud.js [hunk 1]
```diff
@@ -1,40 +1,71 @@
...
+fixture`${FIXTURE_INIT}`.page`${TESTCAFE_PHOTOS_URL}/`
+  .before(async ctx => {
+    ctx.vr = new VisualReviewTestcafe({
+      projectName: `${SLUG}`,
+      suiteName: `${FIXTURE_INIT}`
+    })
+    await ctx.vr.start()
+  })
+  .beforeEach(async t => {
+    console.group(`\n↳ ℹ️  Loggin & Initialization`)
```
Comment (cedricmessiant) [inline testcafe/tests/photos/photos_crud.js]: typo: `Login`

File: testcafe/tests/photos/photos_crud.js [hunk 2]
```diff
@@ -1,40 +1,71 @@
...
+const TEST_VIEWER_FIRST = `1-1 Open viewer for 1st photo`
+const TEST_VIEWER_LAST = `1-1 Open viewer for last photo`
+const TEST_VIEWER_OTHER = `1-1 Open viewer for 2nd photo`
 
-test('Select 1 pic from Photos view', async () => {
-  console.group('↳ ℹ️  Select 1 pic from Photos view')
+fixture`${FIXTURE_INIT}`.page`${TESTCAFE_PHOTOS_URL}/`
+  .before(async ctx => {
+    ctx.vr = new VisualReviewTestcafe({
+      projectName: `${SLUG}`,
```
Comment (cedricmessiant) [inline testcafe/tests/photos/photos_crud.js]: you don't need \`${}\` for those two lines

File: testcafe/tests/photos/photos_crud.js [hunk 3]
```diff
@@ -1,40 +1,71 @@
...
+    await t.useRole(photosUser)
+    await timelinePage.waitForLoading()
+    await timelinePage.initPhotosCount()
+    console.groupEnd()
+  })
+  .after(async ctx => {
+    await ctx.vr.checkRunStatus()
+  })
+
+test(`${TEST_SELECT1}`, async t => {
```
Comment (cedricmessiant) [inline testcafe/tests/photos/photos_crud.js]: Remove \`${}\` (same for all `test()`)

File: testcafe/tests/photos/photos_crud.js [hunk 4]
```diff
@@ -63,14 +97,17 @@ test('Open Last pic', async t => {
...
-  console.group('↳ ℹ️  Open a random pic (not first nor last)')
+test(`${TEST_VIEWER_OTHER}`, async t => {
+  console.group(`↳ ℹ️  ${FEATURE_PREFIX} : ${TEST_VIEWER_OTHER}`)
   //Both arrows show up. Navigatio to other pics is OK, Closing pic (X or 'esc') is Ok
   // We need at least 3 pics in our cozy for this test to pass
   const photoIndex = random(1, t.ctx.totalFilesCount - 2)
 
   console.log('Open random pic  > photoIndex ' + photoIndex)
   await photoViewer.openPhotoFullscreen(photoIndex)
+  await t.fixtureCtx.vr.takeScreenshotAndUpload(
```
Comment (cedricmessiant) [inline testcafe/tests/photos/photos_crud.js]: you could put `t.fixtureCtx.vr.takeScreenshotAndUpload` in a const to ease readability

File: testcafe/tests/photos/photos_crud.js [hunk 5]
```diff
@@ -1,40 +1,71 @@
...
+    `${FEATURE_PREFIX}/${TEST_SELECT2}-1`
+  )
   console.groupEnd()
 })
 
-test('Open 1st pic', async () => {
-  console.group('↳ ℹ️  Open 1st pic')
+test(`${TEST_VIEWER_FIRST}`, async t => {
+  console.group(`↳ ℹ️  ${FEATURE_PREFIX} : ${TEST_VIEWER_FIRST}`)
   //Right arrow shows up. Navigatio to other pics is OK, Closing pic (X or 'esc') is Ok
```
Comment (cedricmessiant) [inline testcafe/tests/photos/photos_crud.js]: typo: Navigation

File: testcafe/tests/photos/photos_crud.js (hunk 2 above)
Comment (CozyKourai) [reply testcafe/tests/photos/photos_crud.js]: Actually this is the old method for initializing VR !
Thanks for pointing it out!
Replace by 
```
   .before(async ctx => {
    await initVR(ctx, SLUG, FIXTURE_INIT)
  })
```

File: testcafe/tests/photos/photos_crud.js (hunk 4 above)
Comment (CozyKourai) [reply testcafe/tests/photos/photos_crud.js]: I don't think so. `t.fixutureCtx` is needed to get the fixture and test context 
https://devexpress.github.io/testcafe/documentation/test-api/test-code-structure.html#sharing-variables-between-fixture-hooks-and-test-code

**Comment by CozyKourai** [comment]
New screenshots, hence visual review on photos (no conflicts with merge screenshots)

File: testcafe/tests/pages/photos/photos-timeline-model.js [hunk 6]
```diff
@@ -31,9 +31,21 @@ export default class Timeline extends Commons {
       )
     await t.takeScreenshot()
     const allPhotosEndCount = await this.getPhotosCount('After')
-    await t
-      .expect(allPhotosEndCount)
-      .eql(t.ctx.totalFilesCount + numOfFiles)
+    await t.expect(allPhotosEndCount).eql(t.ctx.totalFilesCount + numOfFiles)
+  }
+
+  async takeScreenshotsForUpload(screenshotsPath, hasMask = false) {
```
Comment (Crash--) [inline testcafe/tests/pages/photos/photos-timeline-model.js]: Le nom est mal trouvé. et on n'a pas dit qu'on devait passer par la même takeScreenshot quie xiste déjà  ?

Comment (Crash--) [reply testcafe/tests/pages/photos/photos-timeline-model.js]: Il semblerait que c'est djéà plus ou moins fait là : https://github.com/cozy/cozy-drive/pull/1562/files#diff-e2fdf05505f722043b924a300c7e983dR49 ?

Comment (CozyKourai) [reply testcafe/tests/pages/photos/photos-timeline-model.js]: Ah oué c'est le merge, j'ai raté qu'il m'avais remis cette fonction! Je clean tout ca!

**Pull request merged**